tests/Aspire.Hosting.Tests/KestrelConfigTests.cs

```diff
@@ -72,28 +85,72 @@ public void MultipleKestrelHttpEndpointsKeepTheirNames()
+            operation: DistributedApplicationOperation.Run,
+            callback: builder =>
+            {
+                builder.WithHttpEndpoint(5017, name: "ExplicitProxiedHttp");
+                builder.WithHttpEndpoint(5018, name: "ExplicitNoProxyHttp", isProxied: false);
+            });
+
+        // Even though we're in Run mode, we need to get the env in Publish mode since ports don't get allocated
+        var config = await EnvironmentVariableEvaluator.GetEnvironmentVariablesAsync(resource, DistributedApplicationOperation.Publish);
+
+        Assert.Collection(
+            config.Where(envVar => envVar.Key.StartsWith("Kestrel__")),
+            envVar =>
+            {
+                Assert.Equal("Kestrel__Endpoints__FirstHttpEndpoint__Url", envVar.Key);
+                Assert.Equal("http://*:{projectName.bindings.FirstHttpEndpoint.targetPort}", envVar.Value);
+            },
+            envVar =>
+            {
+                Assert.Equal("Kestrel__Endpoints__SecondHttpEndpoint__Url", envVar.Key);
+                // Note that localhost (from Kestrel config) is preserved at runtime
+                Assert.Equal("http://localhost:{projectName.bindings.SecondHttpEndpoint.targetPort}", envVar.Value);
+            },
+            envVar =>
+            {
+                Assert.Equal("Kestrel__Endpoints__ExplicitProxiedHttp__Url", envVar.Key);
+                Assert.Equal("http://*:{projectName.bindings.ExplicitProxiedHttp.targetPort}", envVar.Value);
+            },
+            envVar =>
+            {
+                // Proxy vs no proxy doesn't make a difference here, since we're always dealing with the target port
+                Assert.Equal("Kestrel__Endpoints__ExplicitNoProxyHttp__Url", envVar.Key);
+                Assert.Equal("http://*:{projectName.bindings.ExplicitNoProxyHttp.targetPort}", envVar.Value);
+            }
+            );
+    }
+
     [Fact]
     public async Task VerifyKestrelEndpointManifestGeneration()
     {
         var resource = CreateTestProjectResource<ProjectWithOnlyKestrelHttpEndpoint>();
 
         var manifest = await ManifestUtils.GetManifest(resource);
 
-        var expectedManifest = $$"""
+        var expectedManifest = """
             {
               "type": "project.v0",
               "path": "another-path",
               "env": {
                 "OTEL_DOTNET_EXPERIMENTAL_OTLP_EMIT_EXCEPTION_LOG_ATTRIBUTES": "true",
                 "OTEL_DOTNET_EXPERIMENTAL_OTLP_EMIT_EVENT_LOG_ATTRIBUTES": "true",
                 "OTEL_DOTNET_EXPERIMENTAL_OTLP_RETRY": "in_memory",
-                "ASPNETCORE_FORWARDEDHEADERS_ENABLED": "true"
+                "ASPNETCORE_FORWARDEDHEADERS_ENABLED": "true",
+                "Kestrel__Endpoints__http__Url": "http://*:{projectName.bindings.http.targetPort}",
+                "Kestrel__Endpoints__https__Url": "https://*:{projectName.bindings.https.targetPort}"
```

Comment:
I'm not entirely sure whether it's correct to have https here. It happens because we now use this Kestrel override mechanism for all bindings. This particular binding is a default one that we generate because there are no explicit https bindings. @davidfowl for comments.

Reply:
Actually, I'm sure it shouldn't be there. Fixed.